### PR TITLE
Don't default to MKL on EPYC

### PR DESCRIPTION
**Pull request by ChrisRackauckas**

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -72,6 +73,7 @@ CUDA = "5"
 CUDSS = "0.1, 0.2, 0.3"
 ChainRulesCore = "1.22"
 ConcreteStructs = "0.2.3"
+CpuId = "0.3.2"
```
Comment (chriselrod) [inline Project.toml:76]:
```suggestion
CpuId = "0.3.1"
```
0.3.1 is the latest release.
https://github.com/JuliaRegistries/General/blob/master/C/CpuId/Versions.toml

File: test/static_arrays.jl [hunk 2]
```diff
@@ -19,9 +19,9 @@ for alg in (nothing, LUFactorization(), SVDFactorization(), CholeskyFactorizatio
     @test norm(A * sol .- b) < 1e-10
 
     if __non_native_static_array_alg(alg)
-        @test_broken __solve_no_alloc(A, b, alg)
+        @test_broken __solve_no_alloc(A, b, alg) isa LinearSolution
```
Comment (ChrisRackauckas) [inline test/static_arrays.jl:22]:
```suggestion
        @test_broken __solve_no_alloc(A, b, alg) isa SciMLBase.LinearSolution
```

File: test/static_arrays.jl [hunk 3]
```diff
@@ -19,9 +19,9 @@ for alg in (nothing, LUFactorization(), SVDFactorization(), CholeskyFactorizatio
     @test norm(A * sol .- b) < 1e-10
 
     if __non_native_static_array_alg(alg)
-        @test_broken __solve_no_alloc(A, b, alg)
+        @test_broken __solve_no_alloc(A, b, alg) isa SciMLBase.LinearSolution
     else
-        @test_nowarn __solve_no_alloc(A, b, alg)
+        @test_nowarn __solve_no_alloc(A, b, alg) isa LinearSolution
```
Comment (ChrisRackauckas) [inline test/static_arrays.jl:24]:
```suggestion
        @test_nowarn __solve_no_alloc(A, b, alg) isa SciMLBase.LinearSolution
```

File: test/adjoint.jl [hunk 4]
```diff
@@ -62,7 +62,7 @@ db12 = FiniteDiff.finite_difference_gradient(
 db22 = FiniteDiff.finite_difference_gradient(
     x -> f3(eltype(x).(A), eltype(x).(b1), x), copy(b1))
 
-@test dA≈dA2 atol=1e-4
+@test dA≈dA2 atol=1e-3
```
Comment (ChrisRackauckas) [inline test/adjoint.jl:65]:
```suggestion
@test dA≈dA2 rtol=1e-3
```

File: test/static_arrays.jl [hunk 5]
```diff
@@ -17,13 +17,11 @@ for alg in (nothing, LUFactorization(), SVDFactorization(), CholeskyFactorizatio
     sol = solve(LinearProblem(A, b), alg)
     @inferred solve(LinearProblem(A, b), alg)
     @test norm(A * sol .- b) < 1e-10
-
-    if __non_native_static_array_alg(alg)
-        @test_broken __solve_no_alloc(A, b, alg)
+    if alg isa KrylovJL_GMRES
```
Comment (ChrisRackauckas) [inline test/static_arrays.jl:20]:
```suggestion
    if alg <: KrylovJL_GMRES
```

File: test/static_arrays.jl [hunk 6]
```diff
@@ -17,7 +17,7 @@ for alg in (nothing, LUFactorization(), SVDFactorization(), CholeskyFactorizatio
     sol = solve(LinearProblem(A, b), alg)
     @inferred solve(LinearProblem(A, b), alg)
     @test norm(A * sol .- b) < 1e-10
-    if alg <: KrylovJL_GMRES
```
Comment (oscardssmith) [inline test/static_arrays.jl]:
```suggestion 
    if alg isa KrylovJL_GMRES
```

Comment (ChrisRackauckas) [reply test/static_arrays.jl]: That's what I had first https://github.com/SciML/LinearSolve.jl/pull/518/commits/26574e6e155bf7da8033a60338c7f13ac6546787

Comment (chriselrod) [reply test/static_arrays.jl]: If `a isa T` is like `typeof(a) <: T`, but the former is preferred.
If `a` isn't a type, `a <: T` doesn't make sense and will throw an error.
https://github.com/SciML/LinearSolve.jl/actions/runs/10153821474/job/28077889265?pr=518#step:6:1114

File: test/static_arrays.jl [hunk 7]
```diff
@@ -17,13 +17,11 @@ for alg in (nothing, LUFactorization(), SVDFactorization(), CholeskyFactorizatio
     sol = solve(LinearProblem(A, b), alg)
     @inferred solve(LinearProblem(A, b), alg)
     @test norm(A * sol .- b) < 1e-10
-
-    if __non_native_static_array_alg(alg)
-        @test_broken __solve_no_alloc(A, b, alg)
+    if typeof(alg) <: KrylovJL_GMRES
```
Comment (ChrisRackauckas) [inline test/static_arrays.jl:20]:
```suggestion
    if alg isa Function && alg === KrylovJL_GMRES
```

File: test/static_arrays.jl [hunk 8]
```diff
@@ -17,13 +17,11 @@ for alg in (nothing, LUFactorization(), SVDFactorization(), CholeskyFactorizatio
     sol = solve(LinearProblem(A, b), alg)
     @inferred solve(LinearProblem(A, b), alg)
     @test norm(A * sol .- b) < 1e-10
-
-    if __non_native_static_array_alg(alg)
-        @test_broken __solve_no_alloc(A, b, alg)
+    if alg isa Function && alg === KrylovJL_GMRES
```
Comment (chriselrod) [inline test/static_arrays.jl:20]:
```suggestion
    if alg isa KrylovJL
```
I tested in the REPL:
```julia
julia> using LinearSolve

julia> alg = KrylovJL_GMRES()
KrylovJL{typeof(Krylov.gmres!), Int64, Tuple{}, Base.Pairs{Symbol, Union{}, Tuple{}, @NamedTuple{}}}(Krylov.gmres!, 0, 0, (), Base.Pairs{Symbol, Union{}, Tuple{}, @NamedTuple{}}())

julia> alg isa KrylovJL
true
```

Comment (chriselrod) [inline test/static_arrays.jl:20]:
```suggestion
    if alg isa KrylovJL{typeof(LinearSolve.Krylov.gmres!)}
```
if you'd prefer `gmres!` only
```julia
julia> alg isa KrylovJL{typeof(LinearSolve.Krylov.gmres!)}
true
```

**Pull request merged**